src/transport/PeerAddress.h

```diff
@@ -96,11 +97,15 @@ class PeerAddress
 
     bool IsInitialized() const { return mTransportType != Type::kUndefined; }
 
-    bool operator==(const PeerAddress & other)
+    bool operator==(const PeerAddress & other) const
     {
-        return (mTransportType == other.mTransportType) && (mIPAddress == other.mIPAddress) && (mPort == other.mPort);
+        return (mTransportType == other.mTransportType) && (mIPAddress == other.mIPAddress) && (mPort == other.mPort) &&
+            ((mInterface == INET_NULL_INTERFACEID) || (other.mInterface == INET_NULL_INTERFACEID) ||
```

Comment:
I think we had missed the inteface compare before.

But it seems to make sense to me to have null be treated as 'any' for a compare. If we do not like that, we need more thought into how TCP determines 'an active connection already exists'.

Likely peer communication still needs some thought - we can likely compare IP address however port can be anything since it is random on connection establishment on one side.

Reply:
Ok, updated. 
Our unit tests currently use null interface id, so some bugs may creep up.